src/charm.py

```diff
@@ -80,11 +86,68 @@ def _on_config_changed(self, _):
         else:
             self.unit.status = WaitingStatus("Waiting for Pebble in workload container")
 
+    def _patch_auth_resources(self) -> None:
+        """Patch the Kubernetes auth resources created by Juju.
 
-    def _on_install(self, _):
-        """Placeholder for install event"""
-        pass
+        Raises:
+            ApiError: if patching any of the auth resources fails.
+        """
+        self.unit.status = MaintenanceStatus("Patching auth configuration")
+        client = Client()
+        context = {"namespace": self._namespace, "app_name": self._name}
+
+        with open(Path(self._src_dir) / "auth_manifests.yaml") as f:
+            for obj in codecs.load_all_yaml(f, context=context):
+                try:
+                    client.patch(
+                        type(obj), obj.metadata.name, obj, patch_type=PatchType.MERGE
+                    )
+                except ApiError as e:
+                    logging.error(
+                        f"Patching {obj.metadata.name} failed: {e.status.reason}"
+                    )
+                    raise
+        logging.info("Auth resources successfully patched.")
+
+    def _create_crds(self) -> None:
+        """Creates training-jobs CRDs.
+
+        Raises:
+            ApiError: if creating any of the CRDs fails.
+        """
+        self.unit.status = MaintenanceStatus("Creating CRDs")
+        client = Client()
+
+        with open(Path(self._src_dir) / "crds_manifests.yaml") as f:
+            for obj in codecs.load_all_yaml(f):
+                try:
+                    client.create(obj)
+                    logging.info(f"{obj.metadata.name} CRD successfully created.")
+                except ApiError as e:
+                    if e.status.reason == "AlreadyExists":
+                        logging.info(
+                            f"{obj.metadata.name} CRD already present. It will be used by the operator."
+                        )
+                    else:
+                        logging.error(
+                            f"Creating {obj.metadata.name} failed: {e.status.reason}"
+                        )
+                        raise
 
+    def _on_install(self, _):
+        """Event handler for on-install events."""
+        try:
+            self._patch_auth_resources()
+            self._create_crds()
+        except ApiError as e:
+            logging.error(
+                f"Some Kubernetes resources failed to be created/patched: {str(e)}"
```

Comment:
Can we use `traceback.format_exc()` here instead of `str(e)`? That will log the full traceback, including if there's multiple exceptions involved

Reply:
Done :)